iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/toggle/ResourceGatedByToggledOnFeatureException.java

```diff
@@ -0,0 +1,30 @@
+package com.linkedin.openhouse.internal.catalog.toggle;
+
+/**
+ * The exception thrown when the request feature of a resource is toggled on thus being gated access
+ * or modification.
+ */
+public class ResourceGatedByToggledOnFeatureException extends RuntimeException {
+  public ResourceGatedByToggledOnFeatureException(
+      String featureId, String databaseId, String tableId, Throwable throwable) {
+    this(
+        String.format(
+            "Feature [%s] has been toggled on for the [%s.%s]", featureId, databaseId, tableId),
```

Comment:
It would be very helpful (and with many more features and dimensions to match with, necessary) to also return and log what conditions led to the feature being toggled or not toggled

Reply:
This is the user-side message look like ```
{"status":"INTERNAL_SERVER_ERROR","error":"Internal Server Error","message":"com.linkedin.openhouse.tables.toggle.ResourceGatedByToggledOnFeatureException: Feature [stop_create] has been toggled on for the [demodb.demotable]","stacktrace":null,"cause":"Not Available"}%``` 

Does that look good or you are referring to something different ?

Reply:
whats missing in that message is the full trace of why the feature got toggled on.

for example,

`Feature [stop_create] has been toggled on for the [demodb.demotable] due to predicate [tableId] operand matches [*.*]`

but a short term could be that the operator views the source definition of [stop_create], however with many more features and operand types for the predicate (not limited to tableId, perhaps headless account or client platform) this might become challenging.

Reply:
I see, so the predicate part. 
I think this is enough amount of info to give to end users. Note that this is not dev facing error message. 
But I do plan to make some improvements for exception handling for `ResourceGatedByToggledOnFeatureException` using exception-handler, in a separate PR.